utility/src/test/java/org/oppia/android/util/logging/performancemetrics/PerformanceMetricsAssessorImplTest.kt

```diff
@@ -0,0 +1,281 @@
+import org.oppia.android.util.locale.LocaleProdModule
+import org.oppia.android.util.logging.EnableConsoleLog
+import org.oppia.android.util.logging.EnableFileLog
+import org.oppia.android.util.logging.GlobalLogLevel
+import org.oppia.android.util.logging.LogLevel
+import org.oppia.android.util.logging.SyncStatusModule
+import org.oppia.android.util.networking.NetworkConnectionUtilDebugModule
+import org.oppia.android.util.platformparameter.ENABLE_LANGUAGE_SELECTION_UI_DEFAULT_VALUE
+import org.oppia.android.util.platformparameter.EnableLanguageSelectionUi
+import org.oppia.android.util.platformparameter.LearnerStudyAnalytics
+import org.oppia.android.util.platformparameter.PlatformParameterValue
+import org.oppia.android.util.platformparameter.SPLASH_SCREEN_WELCOME_MSG_DEFAULT_VALUE
+import org.oppia.android.util.platformparameter.SYNC_UP_WORKER_TIME_PERIOD_IN_HOURS_DEFAULT_VALUE
+import org.oppia.android.util.platformparameter.SplashScreenWelcomeMsg
+import org.oppia.android.util.platformparameter.SyncUpWorkerTimePeriodHours
+import org.robolectric.Shadows.shadowOf
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import org.robolectric.shadows.ShadowActivityManager
+import java.io.File
+import javax.inject.Inject
+import javax.inject.Singleton
+
+private const val TEST_PACKAGE_LABEL = "TEST_PACKAGE_LABEL"
+private const val TEST_APP_PATH = "TEST_APP_PATH"
+private const val TEST_PID = 1
+private const val TEST_FILE_NAME = "TEST_FILE_NAME"
+private const val TEST_FILE_CONTENT = "TEST_FILE_CONTENT"
+
+/** Tests for [PerformanceMetricsAssessorImpl]. */
+// FunctionName: test names are conventionally named with underscores.
+@Suppress("FunctionName")
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(application = PerformanceMetricsAssessorImplTest.TestApplication::class)
+class PerformanceMetricsAssessorImplTest {
+
+  @Inject
+  lateinit var performanceMetricsUtilsImpl: PerformanceMetricsAssessorImpl
+
+  @Inject
+  lateinit var context: Context
+
+  @Before
+  fun setUp() {
+    MockitoAnnotations.initMocks(this)
+    setUpTestApplicationComponent()
+  }
+
+  @Test
+  fun testPerformanceMetricsUtils_setTotalMemory_returnsCorrectMemoryTier() {
+    val activityManager: ActivityManager =
+      ApplicationProvider.getApplicationContext<Application>()
+        .getSystemService(Context.ACTIVITY_SERVICE) as ActivityManager
+    val shadowActivityManager: ShadowActivityManager = shadowOf(activityManager)
+    val memoryInfo = ActivityManager.MemoryInfo()
+    memoryInfo.totalMem = (1.5 * 1024 * 1024 * 1024).toLong()
+    shadowActivityManager.setMemoryInfo(memoryInfo)
+    val memoryTier = performanceMetricsUtilsImpl.getDeviceMemoryTier()
+
+    assertThat(memoryTier).isEqualTo(OppiaMetricLog.MemoryTier.LOW_MEMORY_TIER)
+  }
+
+  @Test
+  fun testPerformanceMetricsUtils_getTotalStorageUsed_returnsCorrectStorageUsage() {
```

Comment:
It may be useful to test directory nesting in additional cases for getUsedStorage() to make sure that the implementation is correctly accounting for such cases.

Reply:
done.